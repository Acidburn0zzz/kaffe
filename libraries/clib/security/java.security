#
# Security configuration file for Kaffe
#
security.provider.1=kaffe.security.provider.Kaffe
security.provider.2=sun.security.provider.Sun
