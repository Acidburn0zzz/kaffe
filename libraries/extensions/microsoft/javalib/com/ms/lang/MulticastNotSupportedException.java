package com.ms.lang;

import java.lang.Exception;

public class MulticastNotSupportedException extends Exception {

public MulticastNotSupportedException() {
	super();
}

public MulticastNotSupportedException(String mess) {
	super(mess);
}

}
