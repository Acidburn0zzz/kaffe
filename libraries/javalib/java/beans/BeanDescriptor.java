/*
 * Java core library component.
 *
 * Copyright (c) 1997, 1998
 *      Transvirtual Technologies, Inc.  All rights reserved.
 *
 * See the file "license.terms" for information on usage and redistribution
 * of this file.
 */

package java.beans;

public class BeanDescriptor extends FeatureDescriptor {

  private Class bclass;
  private Class cclass;

  public BeanDescriptor(Class beanClass)
  {
    this(beanClass, null);
  }

  public BeanDescriptor(Class beanClass, Class customizerClass)
  {
    bclass = beanClass;
    cclass = customizerClass;
    /* The default name is the class name without the package name.
     * It is also the default value for display name and short
     * description.
     */
    String name = beanClass.getName();
    name = name.substring(name.lastIndexOf('.') + 1);
    setName(name);
    setDisplayName(name);
    setShortDescription(name);
  }

  public Class getBeanClass()
  {
    return (bclass);
  }

  public Class getCustomizerClass()
  {
    return (cclass);
  }

}
