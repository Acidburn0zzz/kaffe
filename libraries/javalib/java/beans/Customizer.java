/*
 * Java core library component.
 *
 * Copyright (c) 1997, 1998
 *      Transvirtual Technologies, Inc.  All rights reserved.
 *
 * See the file "license.terms" for information on usage and redistribution
 * of this file.
 */

package java.beans;

public interface Customizer {

  public abstract void setObject(Object bean);
  public abstract void addPropertyChangeListener(PropertyChangeListener listener);
  public abstract void removePropertyChangeListener(PropertyChangeListener listener);

}
