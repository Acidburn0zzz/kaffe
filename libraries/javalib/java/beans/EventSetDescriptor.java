package java.beans;

import java.lang.reflect.Method;

/*
 * Java core library component.
 *
 * Copyright (c) 1997, 1998
 *      Transvirtual Technologies, Inc.  All rights reserved.
 *
 * See the file "license.terms" for information on usage and redistribution
 * of this file.
 */
public class EventSetDescriptor
  extends FeatureDescriptor
{
	private final Class sourceClass;
	private final String eventSetName;
	private final Class listenerType;
	private MethodDescriptor[] listenerMethods;
	private Method addListenerMethod;
	private Method removeListenerMethod;
	private boolean unicast;
	private boolean indefault;

public EventSetDescriptor(Class sourceClass, String eventSetName, Class listenerType, String listenerMethodName) throws IntrospectionException
{
	this.sourceClass = sourceClass;
	this.eventSetName = eventSetName;
	this.listenerType = listenerType;

	final String esname = capitalize(eventSetName);
	final String addListenerMethodName = "add"+esname+"Listener";
	final String removeListenerMethodName = "remove"+esname+"Listener";

	listenerMethods = new MethodDescriptor[1];

	// Get the methods on this interface and search out the listener names
	Method meths[] = listenerType.getDeclaredMethods();
	for (int i = 0; i < meths.length; i++) {
		final String mname = meths[i].getName();
		if (mname.equals(listenerMethodName)) {
			listenerMethods[0] = new MethodDescriptor(meths[i]);
		}
		else if (mname.equals(addListenerMethodName)) {
			addListenerMethod = meths[i];
		}
		else if (mname.equals(removeListenerMethodName)) {
			removeListenerMethod = meths[i];
		}
	}
}

public EventSetDescriptor(Class sourceClass, String eventSetName, Class listenerType, String listenerMethodNames[], String addListenerMethodName, String removeListenerMethodName) throws IntrospectionException
{
	this.sourceClass = sourceClass;
	this.eventSetName = eventSetName;
	this.listenerType = listenerType;

	// Get the methods on this interface and search out the listener names
	final Method meths[] = listenerType.getDeclaredMethods();
	listenerMethods = new MethodDescriptor[listenerMethodNames.length];
	for (int i = 0; i < meths.length; i++) {
		final String mname = meths[i].getName();
		for (int j = 0; j < listenerMethodNames.length; j++) {
			if (mname.equals(listenerMethodNames[j])) {
				listenerMethods[j] = new MethodDescriptor(meths[i]);
			}
		}
		if (mname.equals(addListenerMethodName)) {
			addListenerMethod = meths[i];
		}
		else if (mname.equals(removeListenerMethodName)) {
			removeListenerMethod = meths[i];
		}
	}
}

public EventSetDescriptor(String eventSetName, Class listenerType, MethodDescriptor listenerMethodDescriptors[], Method addListenerMethod, Method removeListenerMethod) throws IntrospectionException
{
	this.sourceClass = null;
	this.eventSetName = eventSetName;
	this.listenerType = listenerType;
	this.listenerMethods = listenerMethodDescriptors;
	this.addListenerMethod = addListenerMethod;
	this.removeListenerMethod = removeListenerMethod;
	this.unicast = false;
	this.indefault = false;
	}

public EventSetDescriptor(String eventSetName, Class listenerType, Method listenerMethods[], Method addListenerMethod, Method removeListenerMethod) throws IntrospectionException
{
	this.sourceClass = null;
	this.eventSetName = eventSetName;
	this.listenerType = listenerType;
	this.addListenerMethod = addListenerMethod;
	this.removeListenerMethod = removeListenerMethod;

	this.listenerMethods = new MethodDescriptor[listenerMethods.length];
	for (int i = 0; i < listenerMethods.length; i++) {
		this.listenerMethods[i] = new MethodDescriptor(listenerMethods[i]);
	}

	this.unicast = false;
	this.indefault = false;
	}

public Method getAddListenerMethod()
	{
	return (addListenerMethod);
}

public MethodDescriptor[] getListenerMethodDescriptors()
	{
	return (listenerMethods);
}

public Method[] getListenerMethods()
	{
	Method meths[] = new Method[listenerMethods.length];
	for (int i = 0; i < meths.length; i++) {
		meths[i] = listenerMethods[i].getMethod();
	}
	return (meths);
}

public Class getListenerType()
	{
	return (listenerType);
}

public Method getRemoveListenerMethod()
	{
	return (removeListenerMethod);
}

public boolean isInDefaultEventSet()
	{
	return (indefault);
	}

public boolean isUnicast()
	{
	return (unicast);
}

public void setInDefaultEventSet(boolean inDefaultEventSet)
	{
	indefault = inDefaultEventSet;
	}

public void setUnicast(boolean unicast)
	{
	this.unicast = unicast;
}
}
