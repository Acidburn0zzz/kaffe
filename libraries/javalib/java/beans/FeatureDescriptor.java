/*
 * Java core library component.
 *
 * Copyright (c) 1997, 1998
 *      Transvirtual Technologies, Inc.  All rights reserved.
 *
 * See the file "license.terms" for information on usage and redistribution
 * of this file.
 */

package java.beans;

import java.util.Hashtable;
import java.util.Enumeration;

public class FeatureDescriptor {

  private String name;
  private String dname;
  private boolean expert;
  private boolean hidden;
  private String sdesc;
  private Hashtable table;

  public FeatureDescriptor()
  {
    name = null;
    dname = null;
    expert = false;
    hidden = false;
    sdesc = null;
    table = new Hashtable();
  }

  FeatureDescriptor(String name)
  {
	this();
	this.name = name;
  }

  public String getName()
  {
    return (name);
  }

  public void setName(String name)
  {
    this.name = name;
  }

  public String getDisplayName()
  {
    return (dname);
  }

  public void setDisplayName(String displayName)
  {
    this.dname = displayName;
  }

  public boolean isExpert()
  {
    return (expert);
  }

  public void setExpert(boolean expert)
  {
    this.expert = expert;
  }

  public boolean isHidden()
  {
    return (hidden);
  }

  public void setHidden(boolean hidden)
  {
    this.hidden = hidden;
  }

  public String getShortDescription()
  {
    return (sdesc);
  }

  public void setShortDescription(String text)
  {
    sdesc = text;
  }

  public void setValue(String attributeName, Object value)
  {
    table.put(attributeName, value);
  }

  public Object getValue(String attributeName)
  {
    return (table.get(attributeName));
  }

  public Enumeration attributeNames()
  {
    return (table.elements());
  }

  static String capitalize(String name)
  {
    if (name.length() > 2 && Character.isUpperCase(name.charAt(0)) && Character.isUpperCase(name.charAt(1))) {
      return (name);
    }
    else {
      return (Character.toUpperCase(name.charAt(0)) + name.substring(1));
    }
  }

}
