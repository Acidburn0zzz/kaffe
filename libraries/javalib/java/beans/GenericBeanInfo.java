/*
 * Java core library component.
 *
 * Copyright (c) 1997, 1998
 *      Transvirtual Technologies, Inc.  All rights reserved.
 *
 * See the file "license.terms" for information on usage and redistribution
 * of this file.
 */

package java.beans;

import java.awt.Image;

class GenericBeanInfo extends SimpleBeanInfo {

private BeanDescriptor bean;
private PropertyDescriptor[] properties;
private MethodDescriptor[] methods;
private EventSetDescriptor[] events;
private int defaultEvent;
private int defaultProperty;
private BeanInfo target;

public GenericBeanInfo(BeanDescriptor bean, PropertyDescriptor[] properties, MethodDescriptor[] methods, EventSetDescriptor[] events, int defaultEvent, int defaultProperty, BeanInfo target) {
	this.bean = bean;
	this.properties = properties;
	this.methods = methods;
	this.events = events;
	this.defaultEvent = defaultEvent;
	this.defaultProperty = defaultProperty;
	this.target = target;
}

public BeanDescriptor getBeanDescriptor() {
	return (bean);
}

public int getDefaultEventIndex() {
	return (defaultEvent);
}

public int getDefaultPropertyIndex() {
	return (defaultProperty);
}

public EventSetDescriptor[] getEventSetDescriptors() {
	return (events);
}

public Image getIcon(int i) {
	if (target != null) {
		return (target.getIcon(i));
	}
	else {
		return (null);
	}
}

public MethodDescriptor[] getMethodDescriptors() {
	return (methods);
}

public PropertyDescriptor[] getPropertyDescriptors() {
	return (properties);
}

}
