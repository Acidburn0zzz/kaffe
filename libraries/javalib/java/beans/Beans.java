package java.beans;

import java.io.IOException;
import java.io.InputStream;
import java.io.ObjectInputStream;
import kaffe.io.ClassLoaderObjectInputStream;
import kaffe.lang.SystemClassLoader;

/*
 * Java core library component.
 *
 * Copyright (c) 1997, 1998
 *      Transvirtual Technologies, Inc.  All rights reserved.
 *
 * See the file "license.terms" for information on usage and redistribution
 * of this file.
 */
public class Beans
{
	private static boolean designtime = false;
	private static boolean guiavailable = false;

public Beans() {
}

public static Object getInstanceOf(Object bean, Class targetType) {
	return (bean);
}

public static Object instantiate(ClassLoader cld, String beanName) throws IOException, ClassNotFoundException {
	if (cld == null) {
		cld = SystemClassLoader.getClassLoader();
	}

	InputStream in = cld.getResourceAsStream(beanName.replace('.', '/') + ".ser");

	Object obj;
	if (in != null) {
		obj = (new ClassLoaderObjectInputStream(in, cld)).readObject();
	}
	else {
		try {
			obj = cld.loadClass(beanName).newInstance();
		}
		catch (InstantiationException _) {
			throw new ClassNotFoundException(beanName);
		}
		catch (IllegalAccessException _) {
			throw new ClassNotFoundException(beanName);
		}
	}
	return (obj);
}

public static boolean isDesignTime() {
	return (designtime);
}

public static boolean isGuiAvailable() {
	return (guiavailable);
}

public static boolean isInstanceOf(Object bean, Class targetType) {
	return (targetType.isInstance(bean));
}

public static void setDesignTime(boolean isDesignTime) throws SecurityException {
	// System.getSecurityManager().check???
	designtime = isDesignTime;
}

public static void setGuiAvailable(boolean isGuiAvailable) throws SecurityException {
	// System.getSecurityManager().check???
	guiavailable = isGuiAvailable;
}

}
