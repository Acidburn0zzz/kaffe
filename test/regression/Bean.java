import java.beans.*;
import java.lang.*;
import java.lang.reflect.*;

public class Bean {
  public static void main(String[] args)
  {
    try {
      Class cls = Class.forName("Bean");

      BeanInfo info = Introspector.getBeanInfo(cls);
      System.out.println("BeanDescriptor:");

      System.out.println("  Class: " + info.getBeanDescriptor().getBeanClass());

      MethodDescriptor[] meths = info.getMethodDescriptors();
      for (int i = 0; i < meths.length; i++) {
        System.out.println("  Method: " + meths[i].getMethod());
      }

      PropertyDescriptor[] props = info.getPropertyDescriptors();
      for (int i = 0; i < props.length; i++) {
	PropertyDescriptor prop = props[i];
	System.out.print("  Property: type " + prop.getPropertyType());
        Method rm = prop.getReadMethod();
        Method wm = prop.getWriteMethod();
        System.out.print(", read " + (rm == null ? "<none>" : rm.toString()));
        System.out.print(", write " + (wm == null ? "<none>" : wm.toString()));
	if (prop instanceof IndexedPropertyDescriptor) {
	  IndexedPropertyDescriptor iprop = (IndexedPropertyDescriptor)prop;
          Method rim = iprop.getIndexedReadMethod();
          Method wim = iprop.getIndexedWriteMethod();
          System.out.print(", idx read " + (rim == null ? "<none>" : rim.toString()));
          System.out.print(", idx write " + (wim == null ? "<none>" : wim.toString()));
	}
        System.out.println();

	PropertyEditor editor = PropertyEditorManager.findEditor(prop.getPropertyType());
	System.out.println("  Editor: " + (editor == null ? "<unknown>" : editor.getClass().toString()));
      }

    }
    catch (Exception e) {
      if (e.getMessage() != null) {
        System.out.println(e.getMessage());
      }
      e.printStackTrace();
    }
  }

  // These are here so we can check various properties out.
  public String getValue() { return (null); }
  public void setValue(String val) { }
  public String getArray(int idx) { return (null); }
  public void setArray(int idx, String val) { }
  public String[] getArray() { return (new String[0]); }
  public void setArray(String[] val) { }
}

/* Expected Output:
BeanDescriptor:
  Class: class Bean
  Method: public void Bean.setArray(java.lang.String[])
  Method: public final void java.lang.Object.wait(long,int) throws java.lang.InterruptedException
  Method: public final void java.lang.Object.wait() throws java.lang.InterruptedException
  Method: public void Bean.setArray(int,java.lang.String)
  Method: public boolean java.lang.Object.equals(java.lang.Object)
  Method: public final native void java.lang.Object.notifyAll()
  Method: public void Bean.setValue(java.lang.String)
  Method: public java.lang.String[] Bean.getArray()
  Method: public final native java.lang.Class java.lang.Object.getClass()
  Method: public java.lang.String Bean.getValue()
  Method: public java.lang.String Bean.getArray(int)
  Method: public final void java.lang.Object.wait(long) throws java.lang.InterruptedException
  Method: public native int java.lang.Object.hashCode()
  Method: public static void Bean.main(java.lang.String[])
  Method: public java.lang.String java.lang.Object.toString()
  Method: public final native void java.lang.Object.notify()
  Property: type class java.lang.Class, read public final native java.lang.Class java.lang.Object.getClass(), write <none>
  Editor: <unknown>
  Property: type class [Ljava.lang.String;, read public java.lang.String[] Bean.getArray(), write public void Bean.setArray(java.lang.String[]), idx read public java.lang.String Bean.getArray(int), idx write public void Bean.setArray(int,java.lang.String)
  Editor: <unknown>
  Property: type class java.lang.String, read public java.lang.String Bean.getValue(), write public void Bean.setValue(java.lang.String)
  Editor: class kaffe.beans.editors.StringEditor
*/
