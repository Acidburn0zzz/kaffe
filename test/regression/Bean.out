BeanDescriptor:
  Class: Bean
  Method: public void Bean.setArray(java.lang.String[])
  Method: public final void java.lang.Object.wait(long,int) throws java.lang.InterruptedException
  Method: public final void java.lang.Object.wait() throws java.lang.InterruptedException
  Method: public void Bean.setArray(int,java.lang.String)
  Method: public boolean java.lang.Object.equals(java.lang.Object)
  Method: public final native void java.lang.Object.notifyAll()
  Method: public void Bean.setValue(java.lang.String)
  Method: public java.lang.String[] Bean.getArray()
  Method: public final native java.lang.Class java.lang.Object.getClass()
  Method: public java.lang.String Bean.getValue()
  Method: public java.lang.String Bean.getArray(int)
  Method: public final void java.lang.Object.wait(long) throws java.lang.InterruptedException
  Method: public native int java.lang.Object.hashCode()
  Method: public static void Bean.main(java.lang.String[])
  Method: public java.lang.String java.lang.Object.toString()
  Method: public final native void java.lang.Object.notify()
  Property: type java.lang.Class, read public final native java.lang.Class java.lang.Object.getClass(), write <none>
  Editor: <unknown>
  Property: type java.lang.String[], read public java.lang.String[] Bean.getArray(), write public void Bean.setArray(java.lang.String[]), idx read public java.lang.String Bean.getArray(int), idx write public void Bean.setArray(int,java.lang.String)
  Editor: <unknown>
  Property: type java.lang.String, read public java.lang.String Bean.getValue(), write public void Bean.setValue(java.lang.String)
  Editor: kaffe.beans.editors.StringEditor
